plugin/evm/block.go

```diff
@@ -112,12 +112,12 @@ func readMainnetBonusBlocks() (map[uint64]ids.ID, error) {
 type Block struct {
 	id        ids.ID
 	ethBlock  *types.Block
-	vm        *VM
+	vm        *sharedEvm
 	atomicTxs []*atomic.Tx
 }
 
 // newBlock returns a new Block wrapping the ethBlock type and implementing the snowman.Block interface
-func (vm *VM) newBlock(ethBlock *types.Block) (*Block, error) {
+func (vm *sharedEvm) newBlock(ethBlock *types.Block) (*Block, error) {
```

Comment:
I assume some further changes to `block.go` is needed to fully have the atomic vm separated from the "base" evm that we can share with subnet-evm?